apis/serving/v1alpha1/servingruntime_webhook.go

```diff
@@ -0,0 +1,214 @@
+//Copyright 2021 IBM Corporation
+//
+//Licensed under the Apache License, Version 2.0 (the "License");
+//you may not use this file except in compliance with the License.
+//You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+//Unless required by applicable law or agreed to in writing, software
+//distributed under the License is distributed on an "AS IS" BASIS,
+//WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//See the License for the specific language governing permissions and
+//limitations under the License.
+//
+package v1alpha1
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	kservev1alpha "github.com/kserve/kserve/pkg/apis/serving/v1alpha1"
+	"github.com/kserve/kserve/pkg/constants"
+	"github.com/kserve/modelmesh-serving/controllers/autoscaler"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+//+kubebuilder:webhook:path=/validate-serving-modelmesh-io-v1alpha1-servingruntime,mutating=false,failurePolicy=fail,sideEffects=None,groups=serving.kserve.io,resources=servingruntimes;clusterservingruntimes,verbs=create;update,versions=v1alpha1,name=servingruntime.modelmesh-webhook-server.default,admissionReviewVersions=v1
+type ServingRuntimeWebhook struct {
+	Client  client.Client
+	decoder *admission.Decoder
+}
+
+func (s *ServingRuntimeWebhook) Handle(ctx context.Context, req admission.Request) admission.Response {
+	var srAnnotations map[string]string
+	srReplicas := uint16(65535)
+	multiModel := false
+
+	if req.Kind.Kind == "ServingRuntime" {
```

Comment:
as autoscaling is supported here in ServingRuntime, have you consider to add `minReplicas` and `maxReplicas` into ServingRuntime and deprecated `replicas` field in the future?  as Predictor has similar [fields](https://github.com/kserve/kserve/blob/095ba09cdcda14b8fde3df2b0edc6be40790ce87/pkg/apis/serving/v1beta1/component.go#L76-L91) and ServingRuntime/ClusterServingRuntime is kind of a template for InferenceSerivce, adding might help to support default min/max replicas for inference service as well (if we allow empty min/max replicas in inferenceservice).

Reply:
That is a good idea but we can discuss it later.  for now, it would be enough. If you think it is needed, please create an issue for a future discussion. Thanks!